### PR TITLE
Describe behavior ext. for SFENCE.VMA and HFENCE.VMA

### PR DESCRIPTION
**Pull request by rsahita**
addresses issue #77

### CODE REVIEW

File: chapter3.adoc [hunk 1]
```diff
@@ -244,3 +244,30 @@ prior to the `MINVAL.SPA`, and
 subsequent to the `MFENCE.SPA`
 
 `MINVAL.SPA` is only valid in M-mode.
+
+=== Extension to behavior of SFENCE.VMA and HFENCE.VMA
+
+The behavior of the SFENCE.VMA and the HFENCE.VMA instructions are affected when
```
Comment (ved-rivos) [inline chapter3.adoc:250]: HFENCE.VMA -> HFENCE.GVMA and HFENCE.VVMA.

Should also add a note about their Svinval equivalent.

File: chapter3.adoc [hunk 2]
```diff
@@ -244,3 +244,30 @@ prior to the `MINVAL.SPA`, and
...
+virtual address within the current supervisor domain, and the ASID argument is
+a S/HS-level ASID within the current supervisor domain. The current supervisor
+domain is identified by the SDID field of the CSR mttp, and the effective ASID
+can be considered the combination of the SDID and the S/HS-level ASID. The
+SFENCE.VMA orders stores only to this S/HS-level address-translation structures
+with subsequent HS-level address translations.
+
+When V=1, the virtual-address argument to SFENCE.VMA is a guest virtual address
+within the current virtual machine, and the ASID argument is a VS-level ASID
+within the current virtual machine. The current virtual machine is identified by
```
Comment (ved-rivos) [inline chapter3.adoc:267]:
```suggestion
within the current virtual machine. The current virtual machine is identified by the combination of 
```

File: chapter3.adoc [hunk 3]
```diff
@@ -244,3 +244,30 @@ prior to the `MINVAL.SPA`, and
 subsequent to the `MFENCE.SPA`
 
 `MINVAL.SPA` is only valid in M-mode.
+
+=== Extension to behavior of SFENCE.VMA and HFENCE.VMA
+
+The behavior of the SFENCE.VMA and the HFENCE.VMA instructions are affected when
+supervisor domains are used by the M-mode RDSM.
```
Comment (SiFiveHolland) [inline chapter3.adoc:251]: These fences are affected whenever the `Smsdid` extension is implemented and the mttp.SDID field exists, whether or not M-mode software contains any support for supervisor domains.

```suggestion
the Smsdid extension is implemented.
```

File: chapter3.adoc (hunk 1 above)
Comment (SiFiveHolland) [reply chapter3.adoc:250]: To be thorough, we would need another paragraph for the HFENCE instructions similar to the SFENCE V=0 case saying that "the effective VMID is the combination of the SDID and VMID".

File: chapter3.adoc [hunk 4]
```diff
@@ -244,3 +244,30 @@ prior to the `MINVAL.SPA`, and
 subsequent to the `MFENCE.SPA`
 
 `MINVAL.SPA` is only valid in M-mode.
+
+=== Extension to behavior of SFENCE.VMA and HFENCE.VMA
```
Comment (rsahita) [inline chapter3.adoc:248]:
```suggestion
=== Extension to behavior of SFENCE.VMA, HFENCE.GVMA and HFENCE.VVMA
```

File: chapter3.adoc [hunk 5]
```diff
@@ -244,3 +244,57 @@ prior to the `MINVAL.SPA`, and
...
+HFENCE.VMMA and execute when hgatp.VMID and mttp.SDID have the same setting as
+it did when HFENCE.VVMA executed.
+
+=== Extension to behavior of `Sinval`
+
+When `Sinval` is implemented with `Smsdid` the behavior of SINVAL.VMA is
+modified to also use the SDID from the CSR mttp in addition to the ASID provided
+in SINVAL.VMA.
+
+When the `Hypervisor` extension is also implemented the behavior of HINVAL.VVMA
```
Comment (rsahita) [inline chapter3.adoc:298]:
```suggestion
When the `Hypervisor` extension is also implemented, the behavior of HINVAL.VVMA
```

File: chapter3.adoc [hunk 6]
```diff
@@ -244,3 +244,57 @@ prior to the `MINVAL.SPA`, and
...
+For HFENCE.VVMA, the guest-virtual address argument is within the context of
+the current virtual machine identified by the combination of the ASID specified
+in the vsatp, the VMID field of CSR hgatp, and the SDID field of CSR mttp.
+Executing an HFENCE.VVMA guarantees that any previous stores already visible to
+the current hart are ordered before all implicit reads by that hart done for
+VS-stage address translations for instructions that are subsequent to the
+HFENCE.VMMA and execute when hgatp.VMID and mttp.SDID have the same setting as
+it did when HFENCE.VVMA executed.
+
+=== Extension to behavior of `Sinval`
```
Comment (SiFiveHolland) [inline chapter3.adoc:292]:
```suggestion
=== Extension to behavior of `Svinval`
```

File: chapter3.adoc [hunk 7]
```diff
@@ -244,3 +244,57 @@ prior to the `MINVAL.SPA`, and
...
+
+When `Sinval` is implemented with `Smsdid` the behavior of SINVAL.VMA is
```
Comment (SiFiveHolland) [inline chapter3.adoc:294]:
```suggestion
When `Svinval` is implemented with `Smsdid` the behavior of SINVAL.VMA is
```

**Review by SiFiveHolland — APPROVED** [review]
LGTM. Two more places where Sinval should be Svinval

**Review by ved-rivos — APPROVED** [review]
LGTM. The description turned out to be bit more wordy than needed.

**Comment by rsahita** [comment]
> LGTM. The description turned out to be bit more wordy than needed.

yeah can reduce it in the arc review stage.

**Pull request merged**